src/app/dashboard-widgets/applications-widget/applications-widget.component.ts

```diff
@@ -125,4 +158,10 @@ export class ApplicationsWidgetComponent implements OnDestroy, OnInit {
       this.runBuildConfigs = orderBy(this.runBuildConfigs, ['name']);
     }
   }
+
+  // Navigate to pipelines page
+  private navigateToPipelines(): void {
+    this.router.navigate(['/', this.loggedInUser.attributes.username, this.currentSpace.attributes.name,
```

Comment:
The `loggedInUser` is set via an asynchronous method which happens to involve http calls. Though it would be quite abnormal, there is nothing stopping that http call to be delayed quite a while. And though the path towards calling this function may have a check, it's not immediately obvious that this is safe. This logic applies to `currentSpace` as well.

Best to be safe and null/undefined check these with an appropriate response (maybe notify user something went wrong).

Reply:
I'm going to refactor this to use a routerLink. However, do you have any examples you can point me to in fabric8-ui? 

I see similar navigation in many places. For example, in the header, profile, app overlay, app launcher, etc. In some cases, context?.user?.attributes?.username is used, but none of that provides an error response.